wire-ios-data-model/Source/UseCases/ObserveMLSGroupVerificationStatusUseCase.swift

```diff
@@ -0,0 +1,79 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+public protocol ObserveMLSGroupVerificationStatusUseCaseProtocol {
+
+    func invoke() async
+
+}
+
+public final class ObserveMLSGroupVerificationStatusUseCase: ObserveMLSGroupVerificationStatusUseCaseProtocol {
+
+    // MARK: - Properties
+
+    private let mlsService: MLSServiceInterface
+    private let updateMLSGroupVerificationStatusUseCase: UpdateMLSGroupVerificationStatusUseCaseProtocol
+    private let syncContext: NSManagedObjectContext
+
+    private var epochChangesListenerTask: Task<Void, Error>?
+
+    // MARK: - Life cycle
+
+    public init(
+        mlsService: MLSServiceInterface,
+        updateMLSGroupVerificationStatusUseCase: UpdateMLSGroupVerificationStatusUseCaseProtocol,
+        syncContext: NSManagedObjectContext
+    ) {
+        self.mlsService = mlsService
+        self.updateMLSGroupVerificationStatusUseCase = updateMLSGroupVerificationStatusUseCase
+        self.syncContext = syncContext
+    }
+
+    deinit {
+        epochChangesListenerTask?.cancel()
+    }
+
+    // MARK: - Methods
+
+    public func invoke() async {
+        epochChangesListenerTask = listenForEpochChanges()
+        try? await epochChangesListenerTask?.value
+    }
```

Comment:
I'm a bit confused now. As I understood it, the invoke method should just start the observation and return immediately. So it would look like this:
```suggestion
    public func invoke() {
        epochChangesListenerTask = listenForEpochChanges()
    }
```